pkg/services/govmomi/service.go

```diff
@@ -684,3 +684,45 @@ func (vms *VMService) reconcileClusterModuleMembership(ctx *virtualMachineContex
 	}
 	return nil
 }
+
+// shutdown tries to perform a guest graceful shutdown first before
+// powering off forcibly.
+func (vms *VMService) shutdownVM(ctx *virtualMachineContext) (string, error) {
```

Comment:
I would have expected this func to do the previous poweroff if Hard mode got configured.